packages/formation-react/src/components/EmergencyBanner/EmergencyBanner.js

```diff
@@ -0,0 +1,93 @@
+// Node modules.
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+// Relative imports.
+import AlertBox from '../AlertBox';
+
+const EMERGENCY_BANNER_LOCALSTORAGE = 'EMERGENCY_BANNER';
+
+export class EmergencyBanner extends Component {
+  static propTypes = {
+    content: PropTypes.string.isRequired,
+    recordEvent: PropTypes.func,
+    showClose: PropTypes.bool,
+    title: PropTypes.string.isRequired,
+    type: PropTypes.string.isRequired,
+    visible: PropTypes.bool.isRequired,
+  };
+
+  static defaultProps = {
+    recordEvent: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      dismissed:
+        localStorage.getItem(EMERGENCY_BANNER_LOCALSTORAGE) ===
+        this.prepareEmergencyBannerID(),
+    };
+  }
+
+  prepareEmergencyBannerID = () => `${this.props.title}:${this.props.content}`;
+
+  dismiss = () => {
+    localStorage.setItem(
+      EMERGENCY_BANNER_LOCALSTORAGE,
+      this.prepareEmergencyBannerID(),
+    );
+    this.setState({ dismissed: true });
+  };
+
+  onClick = event => {
+    if (event.target?.nodeName === 'A') {
+      this.props.recordEvent({
+        event: 'nav-warning-alert-box-content-link-click',
+        alertBoxHeading: this.props.title,
+      });
+    }
+  };
+
+  render() {
+    const { showClose, title, content, type, visible } = this.props;
+
+    const { dismissed } = this.state;
+
+    // Escape early if the banner isn't visible or is dismissed.
+    if (!visible || dismissed) {
+      return null;
+    }
+
+    // Derive onCloseAlert depending on the environment.
+    const onCloseAlert = showClose && this.dismiss;
+
+    return (
+      <div
+        onClick={this.onClick}
+        className={classnames(
+          'usa-alert-full-width',
+          'vads-u-border-top--5px',
+          'medium-screen:vads-u-border-top--10px',
+          'vads-c-emergency-banner',
+          {
+            'vads-u-border-color--warning-message': type !== 'error',
+            'vads-u-border-color--secondary': type === 'error',
+          },
+        )}
+        data-e2e-id="emergency-banner"
+      >
+        <AlertBox
+          // eslint-disable-next-line react/no-danger
+          content={<div dangerouslySetInnerHTML={{ __html: content }} />}
+          headline={title}
+          isVisible
+          onCloseAlert={onCloseAlert}
+          status={type}
+        />
+      </div>
+    );
+  }
+}
+
+export default EmergencyBanner;
```

Comment:
I'm not sure if it'll be specified elsewhere anyway but it could be helpful to add a comment about how this is used pretty much exclusively by the homepage right now

Reply:
Oh that's a great idea! Going to add that now 🙂